src/sentry/incidents/endpoints/serializers/workflow_engine_detector.py

```diff
@@ -1,34 +1,339 @@
+from sentry.models.groupopenperiod import GroupOpenPeriod
+from sentry.sentry_apps.models.sentry_app_installation import prepare_ui_component
+from sentry.sentry_apps.services.app import app_service
+from sentry.sentry_apps.services.app.model import RpcSentryAppComponentContext
+from sentry.snuba.models import QuerySubscription
+from sentry.users.models.user import User
+from sentry.users.services.user.model import RpcUser
+from sentry.users.services.user.service import user_service
+from sentry.workflow_engine.models import (
+    Action,
+    ActionGroupStatus,
+    AlertRuleDetector,
+    DataCondition,
+    DataConditionGroupAction,
+    DataSourceDetector,
+    Detector,
+)
+from sentry.workflow_engine.types import DetectorPriorityLevel
 
 
 class WorkflowEngineDetectorSerializer(Serializer):
-    def serialize(self, obj: Detector, attrs, user, **kwargs) -> AlertRuleSerializerResponse:
-        # TODO: Implement this
-        return {
-            "id": "-2",
-            "name": "Test Alert Rule",
-            "organizationId": "-2",
-            "status": 1,
-            "query": "test",
-            "aggregate": "test",
-            "timeWindow": 1,
-            "resolution": 1,
-            "thresholdPeriod": 1,
-            "triggers": [
-                {
-                    "id": "-1",
-                    "status": 1,
-                    "dateModified": datetime.now(),
-                    "dateCreated": datetime.now(),
-                }
+    """
+    A temporary serializer to be used by the old alert rule endpoints to return data read from the new ACI models
+    """
+
+    def __init__(self, expand: list[str] | None = None, prepare_component_fields: bool = False):
+        self.expand = expand or []
+        self.prepare_component_fields = prepare_component_fields
+
+    def add_sentry_app_installations_by_sentry_app_id(
+        self, actions: list[Action], organization_id: int
+    ) -> Mapping[str, RpcSentryAppComponentContext]:
+        sentry_app_installations_by_sentry_app_id: Mapping[str, RpcSentryAppComponentContext] = {}
+        if self.prepare_component_fields:
+            sentry_app_ids = [
+                int(action.config.get("target_identifier"))
+                for action in actions
+                if action.config.get("sentry_app_identifier") is not None
+            ]
+            install_contexts = app_service.get_component_contexts(
+                filter={"app_ids": sentry_app_ids, "organization_id": organization_id},
+                component_type="alert-rule-action",
+            )
+            sentry_app_installations_by_sentry_app_id = {
+                str(context.installation.sentry_app.id): context
+                for context in install_contexts
+                if context.installation.sentry_app
+            }
+        return sentry_app_installations_by_sentry_app_id
+
+    def add_triggers_and_actions(
+        self,
+        result: DefaultDict[Detector, dict[str, Any]],
```

Comment:
it'd be nice to not rely on the side effect of `result` being updated in the nested `add_` methods, instead building the triggers internally on this method and returning that.

basically just:
```py
def add_triggers_and_actions(...):
   result = {}
   # ...
   return result
```
instead of mutating the top level `result` object.